apps/jobs/views.py

```diff
@@ -2775,3 +2775,44 @@ def update_submission_started_at(request, submission_pk):
         return Response(response_data, status=status.HTTP_200_OK)
     else:
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(["PATCH"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def update_submission_data(request, challenge_pk, submission_pk):
```

Comment:
Why do we need a new API? Why can't we simply reuse the `update_submission` API?

Reply:
`update_submission` expects `stdout`, `stderr`, and other submission metadata and updates all those fields. If we just pass status of submission it will clear all other details of a submission. And `PATCH` request is specifically set up for code upload worker it seems as it only allows changing status when submisison is `running`. We can make meta update optional in `PUT` for updating submisison status.